### PR TITLE
*: Refactor validator library

### LINKED ISSUE #11863: Re-evaluate the pkg/validator constructor approach
Placeholder for now. Confirm whether this approach is right going forward to avoid an edge case where the docker validator is run in production environments. The immediate alternative that comes to mind is adding a setup functional option for the validator that will be used. For tests, they would configure this functional pattern; for normal builds, avoid configuring it, and the default binary constructor would be used instead.

### PR DESCRIPTION
**Pull request by timflannagan**
# Description

Fixes #11863. Refactor the xDS validator library to address several potential issues:

- Multiple validator instances defined. Updated implementation DI a single validator instance down to the levels needed to use it (e.g. plugins, RC translator, etc.)
- Avoid the unlikely edge case where the previous validator.New constructor logic fallsback to the docker-based validator in production environments. Remove this type of logic in favor of explicit constructors. The setup package now defaults to the binary-based validator unless tests have explicitly overrided the functional option when constructing a setup instance

# Change Type

/kind cleanup

# Changelog

```release-note
NONE
```

# Additional Notes

### CODE REVIEW

File: internal/kgateway/extensions2/plugins/trafficpolicy/traffic_policy_plugin.go [hunk 1]
```diff
@@ -224,7 +224,7 @@ func registerTypes(ourCli versioned.Interface) {
 	)
 }
 
-func NewPlugin(ctx context.Context, commoncol *common.CommonCollections, mergeSettings string) extensionsplug.Plugin {
+func NewPlugin(ctx context.Context, commoncol *common.CommonCollections, mergeSettings string, v validator.Validator) extensionsplug.Plugin {
```
Comment (ymesika) [inline internal/kgateway/extensions2/plugins/trafficpolicy/traffic_policy_plugin.go:227]: I don't think there is anywhere down the call stack that verifies v is not nil before executing Validate on it, right?
